### PR TITLE
Add Compatibility for WooCommerce Block and Legacy Checkout

### PR DESCRIPTION
**Pull request by wrunivapay**
This pull request introduces several updates to enhance the functionality and development workflow of the UnivaPay for WooCommerce plugin.

Changes

1. [WooCommerce Checkout Block](https://woocommerce.com/document/woocommerce-store-editing/customizing-cart-and-checkout/) Compatibility
2. Set up the necessary configurations and dependencies to facilitate frontend development
3. Bug Fix 
     - Authorization status is not properly setup on order information
     - Capture on Link Form: Resolved an issue where the capture functionality was not working correctly on the link form
4. [Align flow for Legacy and Block Checkout and Link Charge Token with Order ID](https://github.com/univapay/UnivaPay-for-WooCommerce/pull/10)
5. Update to use [Charge Token rather than Univapay token](https://github.com/univapay/UnivaPay-for-WooCommerce/pull/9)

### CODE REVIEW

**Review by kgston — APPROVED** [review]

**Comment by wrunivapay** [comment]
[10up/action-wordpress-plugin-deploy](https://github.com/10up/action-wordpress-plugin-deploy) / [action-wordpress-plugin-build-zip](https://github.com/10up/action-wordpress-plugin-build-zip) with BUILD_DIR will ignores .distignore,
it just easier to build like [this](https://github.com/univapay/UnivaPay-for-WooCommerce/pull/6/commits/e580bbb1313a612f5d3e0c18644c7208c395f0e4#diff-93e50349c3be43fb7623c98f4b019f9405b7ea64f1fe721f9979078dc316d4c4R38)

**Comment by wrunivapay** [comment]
Although it is possible to use artifacts from a [different workflow](https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories), it is simpler to manage everything within the same workflow.

File: UnivaPay-for-WooCommerce/composer.json [hunk 1]
```diff
@@ -20,11 +19,11 @@
             "wp plugin install woocommerce --activate --path=/tmp/wordpress",
             "cp -r . /tmp/wordpress/wp-content/plugins/UnivaPay-for-WooCommerce",
             "rm -rf /tmp/wordpress/wp-content/plugins/UnivaPay-for-WooCommerce/vendor",
-            "phpunit"
+            "WP_ENV=test phpunit"
         ]
     },
     "require": {
-        "php": ">=7.2",
+        "php": ">=7.4",
```
Comment (kgston) [inline UnivaPay-for-WooCommerce/composer.json:26]: Is this aligned with the min version for woo commerce?

Comment (lymsia) [reply UnivaPay-for-WooCommerce/composer.json:26]: @kgston I am assuming that we are upgrading the PHP SDK to the latest version compatible with PHP 7.4. With the current automated test setup, the oldest WooCommerce version I can support is 8.2. [WC doc](https://woocommerce.com/document/update-php-wordpress/).

I would like to add support for older versions, but I will need to set up the tests differently to achieve that, as wp-cli does not provide older WooCommerce versions (afaik).

Comment (wrunivapay) [reply UnivaPay-for-WooCommerce/composer.json:26]: @kgston I am assuming that we are upgrading the PHP SDK to the latest version compatible with PHP 7.4. With the current automated test setup, the oldest WooCommerce version I can support is 8.2. [WC doc](https://woocommerce.com/document/update-php-wordpress/).

I would like to add support for older versions, but I will need to set up the tests differently to achieve that, as wp-cli does not provide older WooCommerce versions (afaik).

Comment (wrunivapay) [reply UnivaPay-for-WooCommerce/composer.json:26]: and here is the current [test matrix](https://github.com/univapay/UnivaPay-for-WooCommerce/blob/main/.github/workflows/main.yml#L13-L16)

Comment (wrunivapay) [reply UnivaPay-for-WooCommerce/composer.json:26]: in the link that i provide for WooCommerce’s doc on section 1 do mention this one
`PHP versions less than 8.1 and MySQL versions less than 5.6 no longer receive active support, many versions are at End Of Life, and are therefore no longer maintained. As such, using outdated and unsupported versions of MySQL and PHP may expose your site to security vulnerabilities. `

Comment (kgston) [reply UnivaPay-for-WooCommerce/composer.json:26]: All right then 👌

**Review by msbomrel — APPROVED** [review]

**Pull request merged**